hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -434,9 +434,19 @@ private void handleBootstrapEvent(WriteMetadataEvent event) {
     if (Arrays.stream(eventBuffer).allMatch(evt -> evt != null && evt.isBootstrap())) {
       // start to initialize the instant.
       final String instant = Arrays.stream(eventBuffer)
-          .filter(evt -> evt.getWriteStatuses().size() > 0)
-          .findFirst().map(WriteMetadataEvent::getInstantTime)
-          .orElse(WriteMetadataEvent.BOOTSTRAP_INSTANT);
+              .filter(evt -> evt.getWriteStatuses().size() > 0)
+              .findFirst().map(WriteMetadataEvent::getInstantTime)
+              .orElse(WriteMetadataEvent.BOOTSTRAP_INSTANT);
+
+      // if currentInstant is pending && bootstrap event instant is empty
+      // reuse currentInstant, reject bootstrap
+      if (this.metaClient.getActiveTimeline().filterInflightsAndRequested().containsInstant(this.instant) && instant
```

Comment:
We need to reload the active timeline to ensure it is the latest one.

Reply:
Addressed. According to the repairs and test cases submitted by previous people, in this case, we only need to reuse the pending instant in `INSERT` mode.